sycl/include/sycl/ext/oneapi/backend/level_zero.hpp

```diff
@@ -125,6 +125,71 @@ inline context make_context<backend::ext_oneapi_level_zero>(
       BackendObject.Ownership == ext::oneapi::level_zero::ownership::keep);
 }
 
+namespace detail {
+inline std::vector<device> get_all_devices(sycl::platform p) {
+  return p.get_devices();
+}
+
+inline std::optional<sycl::device> find_matching_descendent_device(
+    sycl::device d,
+    const backend_input_t<backend::ext_oneapi_level_zero, device>
+        &BackendObject) {
+  if (get_native<backend::ext_oneapi_level_zero>(d) == BackendObject)
+    return d;
+  std::vector<info::partition_property> partition_props =
+      d.get_info<info::device::partition_properties>();
+
+  for (auto pp : partition_props) {
+    if (pp == info::partition_property::partition_by_affinity_domain) {
+      auto sub_devices = d.create_sub_devices<
+          info::partition_property::partition_by_affinity_domain>(
+          info::partition_affinity_domain::next_partitionable);
+      for (auto sub_dev : sub_devices) {
+        if (auto maybe_device =
+                find_matching_descendent_device(sub_dev, BackendObject))
+          return maybe_device;
+      }
+    }
+
+    assert(false && "Unexpected partitioning scheme for a Level-Zero device!");
```

Comment:
Note that PVC devices may support `ext_intel_partition_by_cslice` depending on the driver configuration.  See [sycl_ext_intel_cslice](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_intel_cslice.asciidoc).

Reply:
https://github.com/intel/llvm/blob/735f1ee496a18851efa8c209e234bc2c5c56b18b/sycl/doc/extensions/supported/sycl_ext_intel_cslice.asciidoc#L208-L219 makes me think it'll just "work".

Reply:
I wasn't reading the code closely enough before.  Now that I have ... Yes, I think you are right.  I'll resolve this comment.